src/main/scala/io/qbeast/spark/delta/OTreeIndex.scala

```diff
@@ -62,6 +80,8 @@ case class OTreeIndex(index: TahoeLogFileIndex) extends FileIndex {
         /* modificationTime */ qbeastBlock.modificationTime,
         absolutePath(qbeastBlock.path))
     }.toArray
+    val stagingStats = stagingFiles
```

Comment:
Two concerns that I have (we should resonate them here first):

1. Could the `stagingFiles` be partitioned? If that's the case, maybe we should check it and return a sequence of `PartitionDirectory` directly.
2. Should we find a way of let Delta (or the underlying format) deal with the filtering of those files? Since we are collecting twice the log, maybe we could improve the number of files read at least.

Reply:
It'd leave the part of improving the READ efficiency to another issue

Reply:
Then I will open one, thanks!